### PR TITLE
 Expand the architecture documentation to include more input related docs.

### PR DESCRIPTION
**Pull request by wiwei**
Whereas we these docs previously provided some basic definitions for input system entities, this expands the docs to include more information on controllers, pointers, and focus.

### CODE REVIEW

File: Documentation/Architecture/InputSystem/CoreSystem.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Events can be sent to global listeners. It's possible to register for all input events by using
+the input systems's [IMixedRealityEventSystem](xref:Microsoft.MixedReality.Toolkit.IMixedRealityEventSystem)
+interface. It's recommended to use the [RegisterHandler](xref:Microsoft.MixedReality.Toolkit.IMixedRealityEventSystem.RegisterHandler``1(IEventSystemHandler))
+method for registering for global events - the deprecated Register function will cause listeners
+to get notified of ALL input events, rather than just input events of a particular type
+(where type is defined by the event interface).
+
+Note that [fallback listeners](xref:Microsoft.MixedReality.Toolkit.Input.MixedRealityInputSystem.PushFallbackInputHandler(GameObject))
+are another type of global listeners which are also discouraged because they will receive
+every single input event that hasn't handled elsewhere in the scene.
```
Comment (thalbern) [inline Documentation/Architecture/InputSystem/CoreSystem.md]:
```suggestion
every single input event that hasn't been handled elsewhere in the scene.
```

File: Documentation/Architecture/InputSystem/CoreSystem.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+1. Event is sent to global listeners.
+2. Event is sent to modal dialogs of the focused object.
+3. Event is sent to the focused object.
+4. Event is sent to fallback listeners.
+
+## Device Managers / Data Providers
+
+These entities are responsible for interfacing with lower-level APIs (such as Windows Mixed Reality APIs,
+or OpenVR APIs, for example) and translating data from those systems into ones that fit the MRTK's higher
```
Comment (thalbern) [inline Documentation/Architecture/InputSystem/CoreSystem.md]:
```suggestion
or OpenVR APIs) and translating data from those systems into ones that fit the MRTK's higher
```

File: Documentation/Architecture/InputSystem/CoreSystem.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+level input abstractions. They are responsible for detecting, creating, and managing the lifetime of
+[controllers](ControllersPointersAndFocus.md#controllers).
+
+The basic flow of a device manager involves:
+
+1. The device manager is instantiated by the input system service.
+2. The device manager registers with its underlying system (for example, the Windows Mixed Reality
+   device manager will register for [gesture and interaction events](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs#L651)).
+3. It creates controllers that it discovers from the underlying system (for example, in this running
```
Comment (thalbern) [inline Documentation/Architecture/InputSystem/CoreSystem.md]:
```suggestion
3. It creates controllers that it discovers from the underlying system (for example 
```

File: Documentation/Architecture/InputSystem/CoreSystem.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+   example, the provider could detect the presence of articulated hands)
```
Comment (thalbern) [inline Documentation/Architecture/InputSystem/CoreSystem.md]:
```suggestion
   the provider could detect the presence of articulated hands)
```

File: Documentation/Architecture/InputSystem/ControllersPointersAndFocus.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+1. A controller gets created by a device manager upon detection of a new source (for example, the
+   detects and starts tracking a hand).
+2. In the controller's Update() loop, it calls into its underlying API system.
+3. In the same update loop, it raises input event changes by calling directly into the
+   core input system itself (for example, raising HandMeshUpdated, or HandJointsUpdated).
+
+## Pointers and focus
+
+Pointers are used to interact with game objects. This section describes how pointers are created,
+how they get updated, and how they determine the object(s) that are in focus. This section will
```
[the remark this answers is not in the document]
Comment (wiwei) [reply Documentation/Architecture/InputSystem/ControllersPointersAndFocus.md]: Yasss

File: Documentation/Architecture/InputSystem/ControllersPointersAndFocus.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+[A pointer](xref:Microsoft.MixedReality.Toolkit.Input.SpherePointer) that uses
+[UnityEngine.Physics.OverlapSphere](https://docs.unity3d.com/ScriptReference/Physics.OverlapSphere.html)
+in order to identify the closest [NearInteractionGrabbable](xref:Microsoft.MixedReality.Toolkit.Input.NearInteractionGrabbable)
+object that should be interacted with. Similar to the PokePointer/NearInteractionTouchable functional
+pair, in order to be interactable with the Sphere Pointer, the game object must contain a component that
+is the NearInteractionGrabbable script. This type of pointer/NearInteractionGrabbable combination is used
+to power higher level scripts like the ManipulationHandler (which provides rotate/resize/translate
+capabilities).
+
+## Teleport Pointer
```
[the remark this answers is not in the document]
Comment (wiwei) [reply Documentation/Architecture/InputSystem/ControllersPointersAndFocus.md]: Ah, this was supposed to be "Teleport pointers" (i.e. a section on teleport pointers)

**Pull request merged**